client/question/questionSpec.js

```diff
@@ -41,15 +46,28 @@ describe('question.js', function() {
         expect(typeof error.regExString).toBe('string');
         expect(typeof error.allowMultiline).toBe('boolean');
         expect(Array.isArray(error.feedbackParagraphs)).toBe(true);
+
         error.feedbackParagraphs.forEach(function(paragraph) {
           expect(typeof paragraph.type).toEqual('string');
-          var equalsTypeText = paragraph.type === PARAGRAPH_TYPE_TEXT;
-          var equalsTypeCode = paragraph.type === PARAGRAPH_TYPE_CODE;
-          var equalsTypeSyntaxError =
-              paragraph.type === PARAGRAPH_TYPE_SYNTAX_ERROR;
-          expect(equalsTypeCode || equalsTypeText || equalsTypeSyntaxError)
-              .toBe(true);
+          expect(ALLOWED_PARAGRAPH_TYPES.indexOf(paragraph.type) !== -1).toBe(
+            true);
+          expect(typeof paragraph.content).toEqual('string');
+        });
+      });
+    });
+  });
+
+  describe('SYSTEM_GENERATED_TIPS', function() {
+    it('should have a valid and consistent internal structure', function() {
```

Comment:
What does valid and consistent mean, in this case?

Reply:
I actually copied this from similar strings in this file. I believe it means that each dict contains the correct set of keys, and that the corresponding values are valid.

I've changed the string for all the tests here to be "should have the correct keys and valid values".